presto-main/src/main/java/io/prestosql/execution/SqlStageExecution.java

```diff
@@ -154,6 +169,29 @@ private SqlStageExecution(StageStateMachine stateMachine, RemoteTaskFactory remo
             }
         }
         this.exchangeSources = fragmentToExchangeSource.build();
+
+        if (isEnableDynamicFiltering(stateMachine.getSession())) {
+            addStateChangeListener(newState -> {
+                if ((newState == SCHEDULING_SPLITS || newState == SCHEDULED || newState == RUNNING)) {
```

Comment:
this does not guarantee that dynamic filters are ready even if `newState == RUNNING`

Reply:
Here we just need the guarantee that all tasks of current stage have been scheduled. DynamicFilterService will record the tasks reported here to make sure that all these scheduled tasks report dynamic filters before it considers a dynamic filter summary ready to use.